### ISSUE TITLE
Null or zero results

### ISSUE THREAD

**Issue by jwidhalm** [LouisPlisso/pytomo#6]
When running the tool there are several fields which are either zero or NULL.  Would this be a result of YouTube now running HTTPS?  

![pytomoresults](https://cloud.githubusercontent.com/assets/13022433/8315154/260f0142-19b4-11e5-9faa-bfaa548acc1b.png)

**Comment by LouisPlisso** [comment]
it should not be an issue as the HTTPS is end-to-end, but at application layer data is decrypted

Should take a further look to understand it

**Comment by LouisPlisso** [comment]
@pari685 do you see this also?

**Comment by pari685** [comment]
@LouisPlisso Yes, I see the same thing. The factory.py file keeps truing for different video types and none of them seem to work.

**Comment by jwidhalm** [comment]
What is the resolution to this problem?

**Comment by LouisPlisso** [comment]
I'm taking a look

**Comment by LouisPlisso** [comment]
the fix is in this PR: https://github.com/LouisPlisso/pytomo/pull/7
@pari685 can you take a look

the code in branch fix_null_values if working for me: @jwidhalm can you check, please?
https://github.com/LouisPlisso/pytomo/tree/fix_null_values

**Comment by pari685** [comment]
Ok.. will check it now...

Thanks,
Parikshit
Email: pari685@gmail.com
Cell : +1-816-585-4103

**Comment by pari685** [comment]
It seems to be working now. However, it still does not populate the AS
number. Also, the crawl seems to stop after 10 videos.

Attached is a database and log file.

**Comment by jwidhalm** [comment]
Louis, Thank you for this update.  I did run some tests and see that some of the previously NULL values are now being populated.  However, I still see NULL values for the following:  VideoType, InitialPlaybackBuffer, and RedirectUrl.  A NULL value for RedirectUrl may be expected behavior if the Url was not redirected but I will leave that assessment up to you.  Also BufferingDuration and BufferingDurationAtEnd were all 0.0 values - this may be related to the IntiialPlaybackBuffer being NULL, but again I'll leave that assessment up to you.  Also, I ran several crawls and noticed that each crawl stopped at 10.  Thank you again for your attention to this.  Please let me know if there is any other information I can provide to assist with this.

**Comment by LouisPlisso** [comment]
@jwidhalm it's expected when you have no interruptions, because you have no buffering

the stop after 10 is not expected: I'll try to reproduce

**Comment by LouisPlisso** [comment]
@pari685 the AS matching IPs may have to be updated, but it worked on my personal connection
have you tried from UMKC or a public ISP

also it's better to create a new issue in order not to mix things up

Thanks
Louis

**Comment by LouisPlisso** [comment]
I've created a new issue for the stop after 10: https://github.com/LouisPlisso/pytomo/issues/8

**Comment by LouisPlisso** [comment]
as per AS number, I've created this issue: https://github.com/LouisPlisso/pytomo/issues/9

**Comment by vgramkris** [comment]
Can you please confirm if the following formula can be used to derive the Time to start play?
TimeToStartPlay(sec) = TimeTogetFirstByte + (InitialData x 8 /(InitialRate x 1000))
